robolectric-resources/src/main/java/org/robolectric/res/ResourceRemapper.java

```diff
@@ -1,61 +1,107 @@
 package org.robolectric.res;
 
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
 import java.lang.reflect.Field;
 import java.lang.reflect.Modifier;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.Map;
-import java.util.Set;
 
 public class ResourceRemapper {
-  private Set<Class<?>> seenClasses = new HashSet<>();
-  private int nextInt = 0x70000000;
-
-  public void remapRClass(Class<?> rClass) {
-    if (seenClasses.add(rClass)) {
-      Map<Integer, Integer> mappings = new HashMap<>();
-
-      for (Class<?> innerClass : rClass.getClasses()) {
-        try {
-          String section = innerClass.getSimpleName();
-
-          if (!section.equals("styleable")) {
-            // first renumber ints
-
-            for (Field field : innerClass.getFields()) {
-              int modifiers = field.getModifiers();
-              if (!Modifier.isFinal(modifiers) && Modifier.isStatic(modifiers)) {
-                if (field.getType() == int.class) {
-                  int newValue = nextInt++;
-                  mappings.put(field.getInt(null), newValue);
-                  field.setAccessible(true);
-                  field.set(null, newValue);
-                }
+
+  private BiMap<String, Integer> resIds = HashBiMap.create();
+  private ResourceIdGenerator resourceIdGenerator = new ResourceIdGenerator(0x7F);
+
+  public void remapRClass(boolean allowFinal, Class<?> rClass) {
```

Comment:
Good idea. I've made it a constructor parameter and improved the documentation also